docs/fastdds/dds_layer/core/waitsets/waitsets.rst

```diff
@@ -12,22 +12,27 @@ to notify communication status changes (including arrival of data) to the applic
 This mechanism is wait-based. Its general use pattern is as follows:
 
 * The application indicates which relevant information it wants to get, by means of :ref:`api_pim_condition`
-  objects (:ref:`api_pim_guardcondition`, :ref:`api_pim_statuscondition`, or ReadCondition)
-  and attaching them to a :ref:`api_pim_waitset`.
-* It then waits on that :ref:`api_pim_waitset` until the trigger value of one or several :ref:`api_pim_condition`
-  objects become true.
-* It then uses the result of the wait (i.e., the list of :ref:`api_pim_condition` objects with
+  objects (:ref:`api_pim_guardcondition`, :ref:`api_pim_statuscondition`, or :ref:`api_pim_readcondition`)
+  and attaching them t
+* It then waits on that :ref:`api_pim_waitset` via the |WaitSet::wait-api|
+  call until the trigger value of one or several :ref:`api_pim_condition` objects become true.
+* It then uses the result of the |WaitSet::wait-api| (i.e., the list of :ref:`api_pim_condition` objects with
   trigger_value == true) to actually get the information by calling:
 
-  * |StatusCondition::get_status_changed-api| and then get_<communication_status> on the relevant Entity, when the
-    condition is a StatusCondition and the status changes refer to plain communication status.
-  * |StatusCondition::get_status_changed-api| and then get_datareaders on the relevant Subscriber, when the condition is a
+  * |Entity::get_status_changes-api|, then checking if any of the changes is relevant using the |StatusMask::is_active-api|
+    method on the result and finally calling get_<communication_status> on the relevant Entity, when the condition is a StatusCondition and the status changes refer to plain communication status.
+    Refer to :ref:`dds_layer_core_status` for additional information on the different statuses that can be queried.
+  * |Entity::get_status_changes-api| and then get_datareaders on the relevant Subscriber, when the condition is a
     StatusCondition and the status changes refer to DataOnReaders.
-  * |StatusCondition::get_status_changed-api| and then read/take on the relevant DataReader, when the condition is a
+  * |Entity::get_status_changes-api| and then |DataReader::read-api|/|DataReader::take-api| on the relevant DataReader, when the condition is a
     StatusCondition and the status changes refer to DataAvailable.
-  * Directly read_w_condition/take_w_condition on the DataReader with the :ref:`api_pim_condition` as a parameter,
+  * Directly |DataReader::read_w_condition-api|/|DataReader::take_w_condition-api| on the DataReader with the :ref:`api_pim_condition` as a parameter,
     when it is a ReadCondition
```

Comment:
I am still not understanding clearly these points. I suppose the best is to talk with @MiguelCompany which was the original writer. As far as I understand, he is trying to explain that there are different kinds of statuses being notified (even though this classification is no where to be found as far as I know). In general the active statuses are queried using the `is_active` API. Then, depending on the status, the operation may vary. For example, if [`DataOnReaders`](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/status/status.html#dds-layer-core-status-dataonreaders) is active then the second bullet point applies. If the active status is [`DataAvailable`](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/status/status.html#dataavailable), then the third bullet point applies. The first bullet point is not clear and I do not understand what it is trying to explain. Finally, the fourth bullet point refers to the `read_w_condition`/`take_w_condition` which is not dependent on the Statuses as far as I know.

I think this part needs to be clearer and maybe some subsections could be included to clarify each possible case.

Reply:
In a follow-up PR this section is going to be reworked